master/internal/db/postgres_checkpoints.go

```diff
@@ -0,0 +1,86 @@
+package db
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// FilterForRegisteredCheckpoints gets the deleted checkpoints provided in the model registry.
+func (db *PgDB) FilterForRegisteredCheckpoints(deleteCheckpoints []uuid.UUID) ([]uuid.UUID, error) {
+	var checkpointIDRows []struct {
+		ID uuid.UUID
+	}
+
+	if err := db.queryRows(`
+	SELECT DISTINCT(c.uuid) AS ID FROM checkpoints_view AS c 
+	JOIN model_versions AS mv ON mv.checkpoint_uuid = c.uuid
+	WHERE c.uuid IN (SELECT UNNEST($1::uuid[])); 
+`, &checkpointIDRows, deleteCheckpoints); err != nil {
+		return nil, fmt.Errorf(
+			"querying for all requested delete checkpoints registered in model registry: %w", err)
+	}
+
+	var checkpointIDs []uuid.UUID
+
+	for _, cRow := range checkpointIDRows {
+		checkpointIDs = append(checkpointIDs, cRow.ID)
+	}
+
+	return checkpointIDs, nil
+}
+
+// MarkCheckpointsDeleted updates the provided delete checkpoints to DELETED state.
+func (db *PgDB) MarkCheckpointsDeleted(deleteCheckpoints []uuid.UUID) error {
+	_, err := db.sql.Exec(`UPDATE raw_checkpoints c
+    SET state = 'DELETED'
+    WHERE c.uuid IN (SELECT UNNEST($1::uuid[]))`, deleteCheckpoints)
```

Comment:
we can talk more offline, but let's not use this `unnest` thing going forward. we ran into a really weird perf issue with it in a more complex query.